tests/integration_tests/test_kani.py

```diff
@@ -19,15 +21,23 @@ def test_kani(results_dir):
     """
     Test all Kani proof harnesses.
     """
+    # https://psutil.readthedocs.io/en/latest/#psutil.virtual_memory confusingly enough returns
+    # information about the physical memory. Limiting to 64GB should exclude most consumer hardware,
+    # preventing the test from locking up peoples computers, while still allowing them to force their
+    # way through it with enough swap.
+    if (
+        psutil.swap_memory().total + psutil.virtual_memory().total
+        <= 64 * 1000 * 1000 * 1000
+    ):
+        pytest.skip("kani requires at least 64GB RAM")
```

Comment:
Why do we even consider swap memory for this? Allowing a test to run knowing in advance it will consume swap space, it will most certainly hang the system :)

Reply:
It would be a bit slower. Why would it hang?

Reply:
Because there are other things in the system that need memory. Swap space is used for swapping in/out physical pages based on the active memory set of all the processes in the system. If you occupy all of that from a single process, nothing else will be able to run.

Reply:
Considering swap space allows people to run the test on cheaper systems (using an ssd for swap space should not affect performance too drastically).

Reply:
> Because there are other things in the system that need memory. Swap space is used for swapping in/out physical pages based on the active memory set of all the processes in the system. If you occupy all of that from a single process, nothing else will be able to run.

True but other things also need RAM, we could add some buffer to consider this but I think this is overkill.

Reply:
what do you mean a very small amount of swap space? It highly depends on the system's setup. Swap is not supposed to be memory. Swap is supposed to provide a buffer for the OS to juggle pages around when it runs out of RAM. A good-citizen program should not depend on the presence or not of swap space.

Reply:
It is not ideal to use swap space, but it is likely the only way most contributors can run the test. I think it is reasonable to support this.

Reply:
Right, I agree with that. However, this should come with a big fat message saying that we are going to eat up all the machine's memory and swap space.

If I was a user and did not know about this requirement, I wouldn't be extremely happy that I ran a test which made my machine hang. So, I think we should add a warning and documentation on the `tests/README.md`

Reply:
So currently the test makes peoples machines hang (mine specifically is what triggered this PR), this check should help minimize this by checking if a user's machine has enough memory. But as you mention it can still hang with this check so it would be worth also adding a warning and documentation.

Reply:
Okay lets just go back to "skip this test unless CI" :sweat_smile:

Reply:
In that case, could we add instructions on how someone may run the proofs on their own?

Reply:
Ah, yeah, true, we do have https://github.com/firecracker-microvm/firecracker/blob/main/docs/formal-verification.md#how-to-run-kani-harnesses, but the first sentence needs to be updated